interface/src/avatar/MyAvatar.cpp

```diff
@@ -3793,54 +3815,115 @@ glm::vec3 MyAvatar::scaleMotorSpeed(const glm::vec3 forward, const glm::vec3 rig
     }
 }
 
-glm::vec3 MyAvatar::calculateScaledDirection(){
+// Calculate the world-space motor velocity for the avatar.
+glm::vec3 MyAvatar::calculateScaledDirection() {
     CharacterController::State state = _characterController.getState();
 
     // compute action input
     // Determine if we're head or controller relative...
-    glm::vec3 forward, right;
+    glm::vec3 forward;
+    glm::vec3 right;
 
-    if (qApp->isHMDMode()) {
-        auto handRotation = getOffHandRotation();
-        glm::vec3 controllerForward(0.0f, 1.0f, 0.0f);
-        glm::vec3 controllerRight(0.0f, 0.0f, (getDominantHand() == DOMINANT_RIGHT_HAND ? 1.0f : -1.0f));
-        glm::vec3 transform;
-        switch (getMovementReference()) {
-            case LocomotionRelativeMovementMode::MOVEMENT_HAND_RELATIVE:
-                forward = (handRotation * controllerForward);
-                right = (handRotation * controllerRight);
-                break;
-            case LocomotionRelativeMovementMode::MOVEMENT_HAND_RELATIVE_LEVELED:
-                forward = (handRotation * controllerForward);
-                transform = forward - (glm::dot(forward, Vectors::UNIT_Y) * Vectors::UNIT_Y);
-                if (glm::length(transform) > EPSILON) {
-                    forward = glm::normalize(transform);
-                } else {
-                    forward = Vectors::ZERO;
-                }
-                right = (handRotation * controllerRight);
-                transform = right - (glm::dot(right, Vectors::UNIT_Y) * Vectors::UNIT_Y);
-                if (glm::length(transform) > EPSILON) {
-                    right = glm::normalize(transform);
-                } else {
-                    right = Vectors::ZERO;
-                }
-                break;
-            case LocomotionRelativeMovementMode::MOVEMENT_HMD_RELATIVE:
-            default:
-                forward = IDENTITY_FORWARD;
-                right = IDENTITY_RIGHT;
+    const int movementReference = getMovementReference();
```

Comment:
Unnecessary `const` variables, here and elsewhere.

Reply:
Thanks, fixed in 8c7c91ed6fad981f098953329e4d0d8bf4809812

Reply:
It's maybe a shame that we don't use `const` more though.  I like it because it tells the reader "this is the only line that sets this variable" :)